### ISSUE TITLE
Solution for the global test accuracy of the FedAvg/FedMD consistently hovers around 10%, which seems to deviate from the results reported in the paper. 

### ISSUE THREAD

**Issue by lunan0320** [CityChan/Federated-Hyper-Knowledge-Distillation#2]
In the origin code, 'if self.args.upload_model == True:' is hidden, and there is no hyper-parameter named 'upload_model', which should be added by us.
Also, 'self.global_model.load_state_dict(global_weights)' is needed to take ahead of global_test.

**Comment by CityChan** [comment]
In the paper, model parameters need to be transmitted, that's why args.upload_model == True is set by default. I was trying to run the experiments that communicating only knowledge without model parameters but it was bad, around 10% global accuracy as you got (randomly guess in 10 classes). Therefore, please set  args.upload_model == True.

**Comment by CityChan** [comment]
And the self.global_model.load_state_dict(global_weights) has been moved to the next line after global_weights = average_weights(local_weights). I guess the reason why you got random guess global accuracy because the global model is not updated until training completed, in the original code. Thank you for the correctness.

**Comment by lunan0320** [comment]
Thanks for your reply,

**Comment by CityChan** [comment]
> Thanks for your reply, it's my honor to give some advice. In addition, I would like to ask 'loss2 -= self.kld(Q_pub,global_soft_prediction[idx].to(self.device))' in 'ClientFedMD.py', why does loss use subtraction here? Looking forward to hearing from you.

please change it to   loss2 = self.kld(Q_help.log(),tensor_global_soft_prediction)
